src/Foundation/Application.php

```diff
@@ -143,7 +143,7 @@ public function isUpToDate()
      */
     public function config($key, $default = null)
     {
-        return array_get($this->make('flarum.config'), $key, $default);
```

Comment:
What do you think about this instead? I’m not sure which Iike better.

```
$config = $this->isInstalled() ? $this->make(‘flarum.config’) : [];

return array_get($config, $key, $default);
```

Reply:
Your version looks nice but I thought we could save a call of the `array_get` method if we directly return the `$default` value or did I miss something?

Reply:
What you say is true, but array_get of an empty array that costly?

Reply:
Nevermind. I thought about it and I think your solution is best.

Reply:
Isn't it interesting that the two methods below have very similar checks to this one? If we go this route, they could be removed.

However, I'm not entirely sure that we *should* go this route and would like to discuss. To me, this feels like we're fixing this problem with band-aid (and not for the first time apparently), but not doing anything about the root cause. These things tend to pile up. For this particular instance, I think the problem starts with [Toby's commit](https://github.com/flarum/core/commit/87bf84ef6e1b04d1fa248f7298fb96981251319e), where I would prefer to inject the config values, instead of resolving the configuration from the IoC container. The latter is often a smell that a class has more knowledge than it should.

This needs some more thought... Any input?

Reply:
@franzliedke I agree that we should fix the root cause and inject config values into the CookieFactory (via setters), but I think it's also a good idea for `$app->config()` to return the default value if the app hasn't been installed, rather than crash.

Reply:
Are thinking of setter methods for things like the cookies names (`cookie.name`) setting?